### ISSUE TITLE
Macro executable target are not ignored on archive (signing)

### ISSUE THREAD

**Issue by alexanderwe** [tuist/tuist#5766]
### What happened?

When archiving a project that includes dependencies which declare macros, e.g. https://github.com/pointfreeco/swift-case-paths/blob/main/Package%40swift-5.9.swift#L38, tuist should not sign those targets since this can lead to issues with projects that do not include a sign certificate for macOS targets.

### How do we reproduce it?

1. Include any SPM dependency which has macros set up 
2. Archive your project with a signing certificate that does not support macOS

### Error log

```sh
No signing certificate "Mac App Distribution" found: No "Mac App Distribution" signing certificate matching team ID "XXXXXXXXXXX" with a private key was found. (in target 'CasePathsMacros' from project 'swift-case-paths')
```

### macOS version

14.2.1

### Tuist version

3.38.0

### Xcode version

15.1

**Comment by jslew** [comment]
Getting a slightly different, but probably related problem, when archiving a tuist workspace for a framework  via `xcodebuild` .  In our case, we are not signing at all, using:

``` 
CODE_SIGN_IDENTITY= \
CODE_SIGNING_REQUIRED=NO \
```
...and xcodebuild fails with this error:
```
note: Using codesigning identity override:
error: An empty identity is not valid when signing a binary for the product type 'Command-line Tool'. (in target 'CasePathsMacros' from project 'swift-case-paths')
```
It seems like the fix might be the same? (not signing the targets)

**Comment by pepicrft** [comment]
Hey @jslew. I'm working on fixing it. We need to strip those macros out when embedding dynamic frameworks.

**Comment by jslew** [comment]
Great, thank you, please let me know if you would like me to try anything

**Comment by alexanderwe** [comment]
@alexanderwe I just tested it and I unfortunately do see this issue now on tuist 4.3.1: 

_**duplicate output file '/Users/User/Library/Developer/Xcode/DerivedData/MyProduct/Build/Products/Debug-iphonesimulator/CasePathsMacros' on task: PhaseScriptExecution Copy Swift Macro executable into $BUILT_PRODUCT_DIR /Users/User/Library/Developer/Xcode/DerivedData/MyProduct/Build/Intermediates.noindex/swift-case-paths.build/Development-watchsimulator/CasePaths.build/Script-9D2512177122777335B34209.sh**_

What's also weird, it mentions iPhone and watchOS here. I do get this for any other combination of my destinations and watchsimulator 

So the first path always changes but the second always stays on watch simulator

**Comment by pepicrft** [comment]
Thanks @alexanderwe. I'll further investigate it.

**Comment by alexanderwe** [comment]
@pepicrft If you need anything else from my side, please just let me know. Happy to provide any information that is needed

**Comment by alexanderwe** [comment]
@pepicrft I investigated a little bit further to better find out under which circumstances the duplicate output file issue happens. 

It appears to only happen on `.iOS` platforms when we have an `embedded` `.watchOS` application in, and both, the `.iOS` and the `.watchOS` app targets, are importing the target that makes use of the  `Copy Swift Macro executable into $BUILT_PRODUCT_DIR`  either directly or indirectly by some of their own dependencies. 

So if you build the watchOS app independently it works. If you build the iOS app without the watch app in its target dependencies also fine. Only when both are part of the same dependency tree the issue happens. 


I have created a minimal reproducible sample project which showcases this issue: [multiplatform_app_with_macros_and_embedded_watchos_app.zip](https://github.com/tuist/tuist/files/14734925/multiplatform_app_with_macros_and_embedded_watchos_app.zip)


![graph](https://github.com/tuist/tuist/assets/12934015/ca8f066f-3783-4631-8c6a-c3b577ff8536)
